terratag.go

```diff
@@ -57,26 +68,52 @@ func tagDirectoryResources(dir string, filter string, matches []string, tags str
 		if isSkipTerratagFiles && strings.HasSuffix(path, "terratag.tf") {
 			log.Print("[INFO] Skipping file ", path, " as it's already tagged")
 		} else {
-			perFile := tagFileResources(path, dir, filter, tags, tfVersion, rename)
-			total.Add(perFile)
+			matchWaitGroup.Add(1)
```

Comment:
using wait group to wait for all the go routines to finish processing.
each go routine processes one file.